magrep.c

```diff
@@ -183,6 +183,8 @@ magrep(char *file)
 		char *v = blaze822_chdr(msg, header);
 		if (v)
 			(void)match_value(file, header, v);
+		else
+			(void)match(file, header, 0);
 	}
```

Comment:
would "else if (vflag) printf..." not be easier here?

Reply:
aha but then we have duplicate logic for cflag etc.

We need more test cases for this.

Reply:
Right! This is the point where it will exit if the header is not found.
Without pondering much, it was easier to reach `match()` where the magic happens (see the check on `s`).